tf2_ros/src/tf2_ros/transform_listener.py

```diff
@@ -27,28 +27,29 @@
 
 # author: Wim Meeussen
 
-import roslib; roslib.load_manifest('tf2_ros')
-import rospy
+from rclpy.node import Node
 import tf2_ros
 from tf2_msgs.msg import TFMessage
 
-class TransformListener():
+class TransformListener(Node):
     """
     :class:`TransformListener` is a convenient way to listen for coordinate frame transformation info.
     This class takes an object that instantiates the :class:`BufferInterface` interface, to which
     it propagates changes to the tf frame graph.
     """
-    def __init__(self, buffer):
+    def __init__(self, buffer, name=None):
```

Comment:
Following the above comment:
```
  def __init__(self, buffer, node=None, name=None):
```
check for `node == None` and build a `Node` instance if required.